plugins/opusCodec/src/OpusEncoder.cpp

```diff
@@ -0,0 +1,269 @@
+    }
+
+}
+
+int AthenaOpusEncoder::getComplexity() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_COMPLEXITY(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setComplexity(int complexity)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_COMPLEXITY(complexity));
+
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting complexity to " << complexity << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getBitrate() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_BITRATE(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setBitrate(int bitrate)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_BITRATE(bitrate));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting bitrate to " << bitrate << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getVBR() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_VBR(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setVBR(int vbr)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_VBR(vbr));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting VBR to " << vbr << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getVBRConstraint() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_VBR_CONSTRAINT(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setVBRConstraint(int vbr_const)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_VBR_CONSTRAINT(vbr_const));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting VBR constraint to " << vbr_const << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getMaxBandwidth() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_MAX_BANDWIDTH(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setMaxBandwidth(int maxbw)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_MAX_BANDWIDTH(maxbw));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting max bandwidth to " << maxbw << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getBandwidth() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_BANDWIDTH(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setBandwidth(int bw)
+{
+    assert(_encoder);
+    int ret = opus_encoder_ctl(_encoder, OPUS_SET_BANDWIDTH(bw));
+    if ( ret != OPUS_OK )
+        qCWarning(encoder) << "Error when setting bandwidth to " << bw << ": " << error_to_string(ret);
+}
+
+int AthenaOpusEncoder::getSignal() const
+{
+    assert(_encoder);
+    int ret;
+    opus_encoder_ctl(_encoder, OPUS_GET_SIGNAL(&ret));
+    return ret;
+}
+
+void AthenaOpusEncoder::setSignal(int signal)
+{
```

Comment:
Please put { on same line as function name  CS 4.1.4 https://github.com/kasenvr/project-athena/blob/kasen/core/CODING_STANDARD.md#414-method-definitions-should-have-the-following-form